config/auth_databricks_wif.go

```diff
@@ -0,0 +1,93 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	"github.com/databricks/databricks-sdk-go/config/credentials"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type DatabricksWIFCredentials struct{}
+
+// Configure implements CredentialsStrategy.
+func (d DatabricksWIFCredentials) Configure(ctx context.Context, cfg *Config) (credentials.CredentialsProvider, error) {
+	if cfg.Host == "" || cfg.ClientID == "" || cfg.TokenAudience == "" {
+		return nil, nil
+	}
+
+	supplier := GithubOIDCTokenSupplier{
+		cfg: cfg,
+	}
+
+	// If no supplier can get an IdToken, skip this CredentialsStrategy
+	idToken, err := supplier.GetOIDCToken(ctx, cfg.TokenAudience)
+	if err != nil {
+		return nil, err
+	}
+	if idToken == "" {
+		return nil, nil
+	}
+
+	endpoints, err := oidcEndpoints(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	ts := &databricksWIFTokenSource{
+		ctx:               ctx,
+		jwtTokenSupplicer: &supplier,
+		audience:          cfg.TokenAudience,
+		clientId:          cfg.ClientID,
+		cfg:               cfg,
+		tokenEndpoint:     endpoints.TokenEndpoint,
+	}
+
+	visitor := refreshableVisitor(ts)
+	return credentials.NewOAuthCredentialsProvider(visitor, ts.Token), nil
+}
+
+// Name implements CredentialsStrategy.
+func (d DatabricksWIFCredentials) Name() string {
+	return "databricks-wif"
+}
+
+type databricksWIFTokenSource struct {
+	ctx               context.Context
+	jwtTokenSupplicer *GithubOIDCTokenSupplier
```

Comment:
typo: jwtTokenSupplier?
suggestion: to be slightly more precise, we can call this the "idTokenSupplier" since the returned access token is also a JWT token.